core/src/main/java/com/sequenceiq/cloudbreak/conclusion/step/InfoCollectorConclusionStep.java

```diff
@@ -0,0 +1,183 @@
+package com.sequenceiq.cloudbreak.conclusion.step;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Queue;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.cedarsoftware.util.io.JsonReader;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.common.json.TypedJsonUtil;
+import com.sequenceiq.cloudbreak.domain.stack.StackStatus;
+import com.sequenceiq.cloudbreak.service.stack.InstanceMetaDataService;
+import com.sequenceiq.cloudbreak.service.stackstatus.StackStatusService;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
+import com.sequenceiq.flow.domain.FlowChainLog;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.service.flowlog.FlowChainLogService;
+import com.sequenceiq.flow.service.flowlog.FlowLogDBService;
+
+@Component
+public class InfoCollectorConclusionStep extends ConclusionStep {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(InfoCollectorConclusionStep.class);
+
+    @Inject
+    private FlowLogDBService flowLogDBService;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
+    @Inject
+    private StackStatusService stackStatusService;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @Override
+    public Conclusion check(Long resourceId) {
+        try {
+            collectDebugInfo(resourceId);
+        } catch (Exception e) {
+            LOGGER.warn("Collecting debug info failed", e);
+        }
+        return succeeded();
+    }
+
+    private void collectDebugInfo(Long resourceId) {
+        StringBuilder debugInfo = new StringBuilder("Flow failed for stackId: ").append(resourceId).append(".\n");
```

Comment:
This newline formatting is quite retro. :)
I'm fine for the for the first iteration but might be good to use a wrapper object, do the collection there and print that out with a nice formatting.